tools/c7n_oci/c7n_oci/provider.py

```diff
@@ -21,13 +47,68 @@ class OCI(Provider):
     resource_map = ResourceMap
 
     def initialize(self, options):
+        if options["account_id"] is None:
+            options["oci_config"] = get_oci_profile_session(options).get_config()
         return options
 
+    def _get_subscribed_region(self, options):
+        client = get_oci_profile_session(options).client("oci.identity.IdentityClient")
+        return client.list_region_subscriptions(tenancy_id=options["oci_config"]["tenancy"]).data
+
+    def _is_subscribed_region(self, subscribed_regions, region, policy_collection):
+        for r in subscribed_regions:
+            if r.region_name == region and r.status == "READY":
+                return True
+
+        policy_collection.log.info(
+            (
+                "User's tenancy is not subscribed to the %s region. "
+                "So skipping this region for the policy execution."
+            ),
+            region,
+        )
+        return False
+
     def initialize_policies(self, policy_collection, options):
+        from c7n.policy import Policy, PolicyCollection
+
+        if options.regions:
+            policies = []
+            subscribed_regions = self._get_subscribed_region(options)
+            if "all" in options.regions:
+                regions = [r.region_name for r in subscribed_regions]
+            else:
+                regions = [
+                    r
+                    for r in options.regions
+                    if self._is_subscribed_region(subscribed_regions, r, policy_collection)
+                ]
+            for p in policy_collection:
+                for region in regions:
+                    options_copy = copy.copy(options)
+                    options_copy.region = str(region)
+                    if (
+                        len(options.regions) > 1
+                        or "all" in options.regions
+                        and getattr(options, "output_dir", None)
+                    ):
+                        options_copy.output_dir = join_output(options.output_dir, region)
+                    policies.append(
+                        Policy(
+                            p.data,
+                            options_copy,
+                            session_factory=policy_collection.session_factory(),
+                        )
+                    )
+            return PolicyCollection(policies, options)
+
         return policy_collection
 
     def get_session_factory(self, options):
-        return SessionFactory(profile=options['profile'])
+        global _oci_profile_session
```

Comment:
why the global profile session here?

Reply:
I wanted to avoid the creation of another session object to get the lists of subscribed regions. So retained the copy. Yeah..No need to have a global level session. I have corrected that.

Reply:
Done. Removed global session.